src/lib/io/sol-iio.c

```diff
@@ -902,6 +970,7 @@ sol_iio_close(struct sol_iio_device *device)
     if (device->fd_handler) sol_fd_del(device->fd_handler);
     if (device->fd > -1) close(device->fd);
     if (device->name_fd > -1) close(device->name_fd);
+    if (NULL != device->mount_matrix) free(device->mount_matrix);
```

Comment:
Two things here. First we're not using yoda style in Soletta and you don't need to check if mount_matrix is not null to call free. Remove the check, calling free(NULL) is safe.